src/main/java/seedu/duke/main/CreateName.java

```diff
@@ -1,23 +1,27 @@
 package seedu.duke.main;
 
-import seedu.duke.data.profile.Profile;
 import seedu.duke.data.profile.attributes.Name;
 import seedu.duke.data.profile.utilities.ProfileUtils;
+import seedu.duke.logic.commands.CommandResult;
 import seedu.duke.logic.parser.exceptions.ParamMissingException;
 import seedu.duke.ui.Ui;
 
 /**
  * Creates a Name Class and prompts user for valid name input.
  */
-public class CreateName {
+public class CreateName extends AttributeCreator {
 
-    private Ui ui;
+    public static final String MESSAGE_BYE_DETECTED = "The command word 'bye' is detected.\n "
+            + "Do you wish to exit this program or to set yor name as %s.\n"
```

Comment:
Think if we detect `bye`, we only need to confirm with them whether they want to exit, if yes then exit, if not then we ask them continuously type a new name, so that we don't need to ask if they want to set their name as bye, which is a bit funny haha.

Reply:
because even the user can change their name to `bye` by using `profile n/bye`, I think it is not considered as a bug?

Reply:
yep its a feature